lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -192,6 +190,22 @@ def add_billing_address(post, payment, options)
         post[:billingAddress] = billing_address
       end
 
+      def name_from_payment(billing_address, payment)
+        return unless payment.first_name && payment.last_name
+
+        billing_address[:firstName] = payment.first_name
+        billing_address[:lastName] = payment.last_name
+      end
+
+      def name_from_address(billing_address, options)
+        return unless address = options[:billing_address]
+
+        first_name, last_name = address[:name].split(' ') if address[:name]
```

Comment:
An edge case that would break this logic would be if there were two spaces in the name field - for example `Michael J. Fox` would become `first_name = Michael` and `last_name = J.`, because the shotgun assignment would only take the first two items in the array.

There is actually a helper method for this type of operation in `gateway.rb` called `split_names` !

Reply:
good to know!